website/content/en/docs/faq.md

```diff
@@ -138,6 +138,25 @@ For more documentation on enabling IPv6 with the Amazon VPC CNI, see the [docs](
 Windows nodes do not support IPv6.
 {{% /alert %}}
 
+### Why do I see extra nodes get launched to schedule pending pods that remain empty and are later removed?
+
+You might have a daemonset, userData configuration, or some other workload that applies a taint after a node is provisioned. After the taint is applied, Karpenter will detect that the pod cannot be scheduled to this new node due to the added taint. As a result, Karpenter will provision yet another node. Typically, the original node has the taint removed and the pod schedules to it, leaving the extra new node unused and reaped by emptiness/consolidation. If the taint is not removed quickly enough, Karpenter may remove the original node before the pod can be scheduled via emptiness consolidation. This could result in an infinite loop of nodes being provisioned and consolidated without the pending pod ever scheduling.
```

Comment:
Don't need to update this in this statement, but I'd also just briefly mention that you probably don't want to be adding taints after the fact after node creation. You're sort-of relying on the fact that you can be fast enough to beat pods to scheduling. Maybe CNI is fine because the node itself won't go ready without the networking up and ready, but generally speaking you want these taints guaranteed to be there on node creation so that you are just responsible for the removal, not the addition

Reply:
Yep, we had the same conversation and conclusion internally when I understood what was going on.